extensions/resend/src/contacts.tsx

```diff
@@ -0,0 +1,258 @@
+  const [audiences, setAudiences] = useState<Audience[]>([]);
+  const [isLoadingContacts, setIsLoadingContacts] = useState<boolean>(false);
+  const [contacts, setContacts] = useState<Contact[]>([]);
+  const [error, setError] = useState("");
+
+  const { isLoading: isLoadingAudience } = useCachedPromise(() => getAudiences(), [], {
+    onData: (data) => {
+      if (data && "data" in data) {
+        setAudiences(data.data);
+      }
+    },
+    onError: (error) => {
+      if (error.name === "validation_error" || error.name === "restricted_api_key") {
+        setError(error.message);
+      }
+    },
+  });
+
+  useEffect(() => {
+    if (audience) {
+      setIsLoadingContacts(true);
+
+      const { id } = audience;
+      const contactsData: Promise<ErrorResponse | GetContactsResponse> = getContacts(id);
+
+      contactsData
+        .then((response) => {
+          if ("statusCode" in response) {
+            // The response is an ErrorResponse
+            setError(`Error ${response.statusCode}: ${response.message}`);
+          } else {
+            // The response is a GetContactsResponse
+            if (response && "data" in response) {
+              setContacts(response.data);
+            }
+          }
+        })
+        .catch((error) => {
+          if (error instanceof Error) {
+            setError(error.message);
+          } else {
+            // The error is of an unknown type or structure
+            setError("An unknown error occurred");
+          }
+        })
+        .finally(() => {
+          setIsLoadingContacts(false);
+        });
+    }
+  }, [audience]);
+
+  async function getContactsFromApi(audienceId: string) {
+    setIsLoadingContacts(true);
+
+    const response = await getContacts(audienceId);
+    if (!("statusCode" in response)) {
+      setContacts(response.data);
+    } else if (response.name === "validation_error" || response.name === "restricted_api_key") {
+      setError(response.message);
+    }
+
+    setIsLoadingContacts(false);
+  }
+
+  async function confirmAndDelete(audienceId: string, contact: Contact) {
+    if (
+      await confirmAlert({
+        title: `Delete '${contact.email}'?`,
+        message: `id: ${contact.id}`,
+        primaryAction: { title: "Delete", style: Alert.ActionStyle.Destructive },
+      })
+    ) {
+      const response = await deleteContact(audienceId, contact.id);
+      if (!("statusCode" in response)) {
+        await showToast(Toast.Style.Success, "Deleted Domain");
+        await getContactsFromApi(audienceId);
+      }
+    }
+  }
+
+  return error ? (
+    <ErrorComponent error={error} />
+  ) : (
+    <List
+      isLoading={isLoadingAudience || isLoadingContacts}
+      searchBarAccessory={<AudienceDropdown audiences={audiences} setAudience={setAudience} />}
+      actions={
+        <ActionPanel>
+          {audience && <Action.Push title="Create Contact" target={<CreateContact audience={audience} />} />}
```

Comment:
Missing Icon.

Reply:
Also, for consistency, maybe you could follow the way I've done it and automatically reload the Contacts list after a new Contact is added? e.g. [Line 98 from api_keys.tsx](https://github.com/raycast/extensions/blob/5bb43b534e7ab265cdfe370c09d88f68c142c5a5/extensions/resend/src/api_keys.tsx#L98). Right now, I need to re-run the entire command, fetch the Audiences and Contact again, in order to see the new contact.